File: MyAFBase/MyAFBase/Utils/PCSChecklist.swift
import Foundation

struct PCSChecklistItem: Identifiable, Equatable {
    let id: String
    let title: String
    let detail: String?
}

enum PCSChecklistKind: String, CaseIterable, Identifiable {
    case inbound
    case outbound

    var id: String { rawValue }

    var title: String {
        switch self {
        case .inbound: "Inbound PCS"
        case .outbound: "Outbound PCS"
        }
    }

    var footer: String {
        switch self {
        case .inbound:
            "Track key in-processing steps for your arrival. Progress is saved on this device."
        case .outbound:
            "Track out-processing and move tasks before your next assignment."
        }
    }

    var systemImage: String {
        switch self {
        case .inbound: "checklist"
        case .outbound: "suitcase"
        }
    }
}

enum PCSChecklist {
    static func items(for kind: PCSChecklistKind) -> [PCSChecklistItem] {
        switch kind {
        case .inbound: inboundItems
        case .outbound: outboundItems
        }
    }

    private static let inboundItems: [PCSChecklistItem] = [
        PCSChecklistItem(id: "in-orders", title: "Obtain and review PCS orders", detail: "Keep digital and paper copies accessible."),
        PCSChecklistItem(id: "in-sponsor", title: "Contact your sponsor", detail: "Confirm reporting date, lodging, and arrival instructions."),
        PCSChecklistItem(id: "in-lodging", title: "Arrange temporary lodging", detail: "Book TLF or approved lodging if arriving before housing."),
        PCSChecklistItem(id: "in-report", title: "Report to unit / MPF", detail: "Complete in-processing within required timelines."),
        PCSChecklistItem(id: "in-deers", title: "Update DEERS and ID card", detail: "Visit MPF or RAPIDS if dependents or cards changed."),
        PCSChecklistItem(id: "in-finance", title: "Visit finance / RAO", detail: "Review pay, BAH, and travel voucher status."),
        PCSChecklistItem(id: "in-medical", title: "Enroll in medical and dental", detail: "Tricare enrollment and PCM assignment."),
        PCSChecklistItem(id: "in-housing", title: "Apply for housing or sign lease", detail: "On-base housing office or off-base lease."),
        PCSChecklistItem(id: "in-vehicle", title: "Register vehicle on installation", detail: "Base registration and required insurance proof."),
        PCSChecklistItem(id: "in-tmo", title: "Coordinate POV / HHG shipment", detail: "Schedule TMO pickup or delivery windows."),
        PCSChecklistItem(id: "in-schools", title: "Enroll children in school / CDC", detail: "If applicable, contact school liaison or CDC."),
        PCSChecklistItem(id: "in-address", title: "Update mailing address and agencies", detail: "USPS, bank, Tricare, and employer records.")
    ]

    private static let outboundItems: [PCSChecklistItem] = [
        PCSChecklistItem(id: "out-orders", title: "Receive and verify outbound orders", detail: "Confirm RNLTD, projected departure, and dependents."),
        PCSChecklistItem(id: "out-mpf", title: "Schedule MPF out-processing", detail: "DEERS updates, assignments, and personnel records."),
        PCSChecklistItem(id: "out-medical", title: "Complete medical and dental clearing", detail: "Records transfer and any required appointments."),
        PCSChecklistItem(id: "out-finance", title: "Finance clearing and final LES review", detail: "Final entitlements, travel vouchers, and advances."),
        PCSChecklistItem(id: "out-housing", title: "Schedule housing move-out inspection", detail: "On-base turn-in or lease termination off-base."),
        PCSChecklistItem(id: "out-cif", title: "Turn in issued equipment / CIF", detail: "Clear supply and equipment accounts."),
        PCSChecklistItem(id: "out-tmo", title: "Schedule TMO / HHG pickup", detail: "Weight tickets and shipment dates."),
        PCSChecklistItem(id: "out-utilities", title: "Cancel utilities and local services", detail: "Power, internet, gym, and other recurring services."),
        PCSChecklistItem(id: "out-gain-base", title: "Research your gaining installation", detail: "Switch bases in MyAFBase to preview your next assignment."),
        PCSChecklistItem(id: "out-pass", title: "Return passes and update gate access", detail: "Visitor passes, long-term decals, and vehicle registration.")
    ]
}

File: android/app/src/main/java/com/ryanladuca/myafbase/domain/logic/PCSChecklist.kt
package com.ryanladuca.myafbase.domain.logic

data class PCSChecklistItem(
    val id: String,
    val title: String,
    val detail: String?
)

enum class PCSChecklistKind(val raw: String, val title: String, val footer: String) {
    INBOUND(
        "inbound",
        "Inbound PCS",
        "Track key in-processing steps for your arrival. Progress is saved on this device."
    ),
    OUTBOUND(
        "outbound",
        "Outbound PCS",
        "Track out-processing and move tasks before your next assignment."
    )
}

object PCSChecklist {
    fun items(forKind: PCSChecklistKind): List<PCSChecklistItem> = when (forKind) {
        PCSChecklistKind.INBOUND -> inboundItems
        PCSChecklistKind.OUTBOUND -> outboundItems
    }

    private val inboundItems = listOf(
        PCSChecklistItem("in-orders", "Obtain and review PCS orders", "Keep digital and paper copies accessible."),
        PCSChecklistItem("in-sponsor", "Contact your sponsor", "Confirm reporting date, lodging, and arrival instructions."),
        PCSChecklistItem("in-lodging", "Arrange temporary lodging", "Book TLF or approved lodging if arriving before housing."),
        PCSChecklistItem("in-report", "Report to unit / MPF", "Complete in-processing within required timelines."),
        PCSChecklistItem("in-deers", "Update DEERS and ID card", "Visit MPF or RAPIDS if dependents or cards changed."),
        PCSChecklistItem("in-finance", "Visit finance / RAO", "Review pay, BAH, and travel voucher status."),
        PCSChecklistItem("in-medical", "Enroll in medical and dental", "Tricare enrollment and PCM assignment."),
        PCSChecklistItem("in-housing", "Apply for housing or sign lease", "On-base housing office or off-base lease."),
        PCSChecklistItem("in-vehicle", "Register vehicle on installation", "Base registration and required insurance proof."),
        PCSChecklistItem("in-tmo", "Coordinate POV / HHG shipment", "Schedule TMO pickup or delivery windows."),
        PCSChecklistItem("in-schools", "Enroll children in school / CDC", "If applicable, contact school liaison or CDC."),
        PCSChecklistItem("in-address", "Update mailing address and agencies", "USPS, bank, Tricare, and employer records.")
    )

    private val outboundItems = listOf(
        PCSChecklistItem("out-orders", "Receive and verify outbound orders", "Confirm RNLTD, projected departure, and dependents."),
        PCSChecklistItem("out-mpf", "Schedule MPF out-processing", "DEERS updates, assignments, and personnel records."),
        PCSChecklistItem("out-medical", "Complete medical and dental clearing", "Records transfer and any required appointments."),
        PCSChecklistItem("out-finance", "Finance clearing and final LES review", "Final entitlements, travel vouchers, and advances."),
        PCSChecklistItem("out-housing", "Schedule housing move-out inspection", "On-base turn-in or lease termination off-base."),
        PCSChecklistItem("out-cif", "Turn in issued equipment / CIF", "Clear supply and equipment accounts."),
        PCSChecklistItem("out-tmo", "Schedule TMO / HHG pickup", "Weight tickets and shipment dates."),
        PCSChecklistItem("out-utilities", "Cancel utilities and local services", "Power, internet, gym, and other recurring services."),
        PCSChecklistItem("out-gain-base", "Research your gaining installation", "Switch bases in MyAFBase to preview your next assignment."),
        PCSChecklistItem("out-pass", "Return passes and update gate access", "Visitor passes, long-term decals, and vehicle registration.")
    )
}
